### PR TITLE
Fixup various nix and CHaP issues

### PR DESCRIPTION
**Pull request by michaelpj**
Still building this, but it looks promising.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -39,6 +39,8 @@
       supportedSystems = import ./supported-systems.nix;
       defaultSystem = head supportedSystems;
 
+      inputMap = { "https://input-output-hk.github.io/cardano-haskell-packages" = CHaP; };
```
Comment (angerman) [inline flake.nix:42]: Maybe we should be more explicit how this works?

We want to ensure that we do get a pinned version of CHaP (as we want to minimise the risk of someone
impersonating GitHub, and subtly giving us some modified CHaP). Thus we use `nix` to lock the CHaP reference,
and then need to pass that locked reference to haskell.nix

1. We reference iog/cardano-haskell-packags in the flake and give it the name `CHaP`.
2. We need to inform haskell.nix about this custom hackage. For this we tell haskell.nix to map the `repository` to the location of `CHaP` from the flake. We record this mapping in the `inputMap`
3. Finally we need pass that `inputMap` to haskell.nix, so it knows where to find the custom hackage.

**Review by angerman — APPROVED** [review]

File: cabal.project [hunk 2]
```diff
@@ -120,26 +92,6 @@ allow-newer:
 -- --------------------------- 8< --------------------------
 -- Please do not put any `source-repository-package` clause above this line.
 
-source-repository-package
-  type: git
-  location: https://github.com/input-output-hk/cardano-node
```
Comment (kderme) [inline cabal.project:125]: This changes the pinned cardano-node version to 1.35.3 though right? My idention was to use node 1.35.4 and the same version dependencies as node 1.35.4. I think this cannot be achived simply by the index-state date, can it?

Comment (michaelpj) [reply cabal.project:125]: You can just use normal version bounds to get that, I think.

**Comment by michaelpj** [comment]
@kderme feel free to take what you like from this, e.g. you can revert the one removing the node s-r-p if you want to do it that way.

**Comment by kderme** [comment]
Many thanks. Though I get this very strange error

**Comment by michaelpj** [comment]
What command are you running?

**Comment by kderme** [comment]
Any nix-build and nix-shell command like
`nix-build -A cardano-db-sync -o db-sync-node`

**Comment by michaelpj** [comment]
Yikes. You appear to have two parallel setups in `flake.nix` and `default.nix`, that's very surprising. I'll fix up `default.nix` also.

**Comment by michaelpj** [comment]
Done, I think.

**Comment by kderme** [comment]
I cherry-picked most of this here https://github.com/input-output-hk/cardano-db-sync/pull/1332. Thanks!

**Pull request closed without merging**